app/assets/javascripts/Components/submission_file_manager.jsx

```diff
@@ -73,6 +77,7 @@ class SubmissionFileManager extends React.Component {
   };
 
   handleDeleteFile = (fileKeys) => {
+    this.setState({viewFile: null, viewFileType: null});
```

Comment:
This one could be merged into the first `then` below. This is a bit trickier because you can't merge the setState into `this.props.onChange` (since it's a prop). You could pass in `this.props.onChange` as a callback to `setState` for that case.